File: Lambda/Alda/translate.py
import logging
from enum import Enum

logger = logging.getLogger()
logger.setLevel(logging.INFO)


class Language(Enum):
    SPANISH = 1


class Translate():

    _language = Language.SPANISH
    _sp_dictionary = {
        'account': 'Cuenta',
        'bonus': 'Bonificación',
        'card': 'Tarjeta',
        'checking': 'Cuenta',
        'credit_card': 'Tarjeta',
        'debit_card': 'Tarjeta',
        'ewallet': 'Billetera',
        'insurance': 'Seguro',
        'investment': 'Inversión',
        'mortgage': 'Hipoteca',
        'savings': 'Ahorros',
        'loan': 'Préstamo'
    }

    def __init__(self, language):
        """
        :param language: Language(ENUM)
        :init: Initializes translation module
        """

    def getTranslation(self, word):
        """ Translated word into _language

        Args:
            word: String, the word to be translated

        Returns:
            The in Translate._language translated word

        To Improve:
            - error when word cannot be translated (is not contained in _language_dictionary)
        """

        if word in self._sp_dictionary:
            return self._sp_dictionary[word]
        else:
            return word

File: Lambda/Alda/lib/translate.js
const spDictionary = {
    'account': 'Cuenta',
    "Bills": "Facturas",
    'bonus': 'Bonificación',
    'card': 'Tarjeta',
    'checking': 'Cuenta',
    'credit_card': 'Tarjeta',
    'debit_card': 'Tarjeta',
    'Education': "Educación",
    'ewallet': 'Billetera',
    "Fitness": "Deportes",
    "Food": "Alimentación",
    'insurance': 'Seguro',
    'investment': 'Inversión',
    "Health": "Salud",
    "Kids": "Niños",
    'mortgage': 'Hipoteca',
    "Pets": "Mascotas",
    "Restaurants": "Restaurantes",
    'savings': 'Ahorros',
    'Shopping': 'Compras',
    "Travel": "Viajes",
    'loan': 'Préstamo'
};

export const translateToSp = (word) => {
    if(spDictionary[word]) {
        return spDictionary[word];
    } else {
        return word;
    }
};
